.github/workflows/test_on_push.yml

```diff
@@ -31,7 +31,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest]
```

Comment:
Because it takes a long for checking in macOS, so we removed that. Now we are adding it back.

Reply:
This is because we were developing the code in a private repository and GitHub Actions time for MacOS runners costs 10 times as much as linux runners and 5 times as much as windows ones. Due to the schedule testing every night, our own pushes to the repo and the length of the test suite, back in June or July, our repo consumed on its own around 80% of all Imperial College quota of GitHub Actions. So we disabled until it was time to open PR to the main PyBaMM repo.

Reply:
Fair enough!